test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```diff
@@ -184,11 +199,17 @@ public void stop() {
 
     public void start(final boolean wait, final ServiceSpec serviceSpec) {
         try {
-            ServiceCreateResponse serviceCreateResponse = Exceptions.handleInterruptedCall(() -> dockerClient.createService(serviceSpec));
+            String serviceId = getID();
+            if (serviceId != null) {
+                 Service service = Exceptions.handleInterruptedCall(() -> dockerClient.inspectService(serviceId));
+                 Exceptions.handleInterrupted(() -> dockerClient.updateService(serviceId, service.version().index(), serviceSpec));
+            } else {
+                ServiceCreateResponse serviceCreateResponse = Exceptions.handleInterruptedCall(() -> dockerClient.createService(serviceSpec));
+                assertNotNull(serviceCreateResponse.id());
```

Comment:
Can we use `org.junit.Assert#assertNotNull(java.lang.String, java.lang.Object)`? This will simplify debugging.

Reply:
Fixed